src/test/java/com/codeninjas/productservice/service/ProductServiceImplTest.java

```diff
@@ -54,7 +54,7 @@ public void updateProduct() throws ProductServiceException {
     }
 
     @Test
-    public void deleteProduct() {
+    public void deleteProduct() throws ProductServiceException {
```

Comment:
can you please remove the public access modifier and rename the method to givenDeleteProductIsSuccess

and also add another unit test void givenDeleteProductThrowsException 

Mockito.when(productRepository.deleteById(any())).thenReturn(Optional.empty());

assertThrows(ProductServiceException.class, () -> { productService.deleteProduct(anyString()); });

Please do the same thing for updateProduct unit tests as well.